### DISCUSSION TITLE
Icon issue in Dock

### DISCUSSION THREAD

**Discussion by MUCCHU** [zed-industries/zed#16398] [Feature Requests]
### Description
Whenever I open Zed either from menu or from the pinned icon in dock, a new icon appears on the dock with the gear icon which is associated with the window that opens(last icon in the screenshot). When I press Alt+tab, I see the proper zed icon and not the gear icon.

### Environment 
OS: Linux Mint 21.3 Cinnamon
Dock: Plank 0.11.89

![image](https://github.com/user-attachments/assets/ac899455-e10a-45d6-945a-535140e67b50)

**Comment by dachinat** [comment]
I have the same problem.

Desktop application entry has correct icon, but an icon for currently opened windows for Zed defaults to window with no icon.

<img width="639" height="580" alt="image" src="https://github.com/user-attachments/assets/219a4e1a-b2c6-48a1-84a4-5e1cff4e1f2f" />

I'm on Arch with MATE desktop environment.

**Reply by dachinat** [reply]
Adding `StartupWMClass=dev.zed.Zed` to a destkop entry and running `update-desktop-database ~/.local/share/applications/
` solves this issue in a way that an opened window icon does not appear anymore, instead there are dots under Zed icon now.

<img width="464" height="97" alt="Screenshot at 2025-08-15 01-18-51" src="https://github.com/user-attachments/assets/2f31f41a-bc26-4d48-9575-bbe26ac809d4" />

**Reply by fndpc** [reply]
Thank u!

**Comment by JosephTLyons** [comment]
Thanks for reaching out. We want to better support our community, and the best way for us to do that is through our official Discord. The team is active there and happy to help.

We’re closing the “Help and General Q&A” category on GitHub Discussions and directing all support questions to Discord going forward.

https://zed.dev/community-links
